fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2093,29 +2094,64 @@ public void boundaryPrimaryKeys() throws Exception {
-                assertEquals(boundaryPrimaryKeys.stream().sorted().distinct().collect(Collectors.toList()), boundaryPrimaryKeys,
-                        "the list should be sorted without duplication.");
-                for (Tuple boundaryPrimaryKey : boundaryPrimaryKeys) {
-                    assertEquals(1, boundaryPrimaryKey.size(), "primary keys should be a single value");
-                }
-            }
+                    .build();
+
+            boundaryPrimaryKeys = recordStore.context.asyncToSync(FDBStoreTimer.Waits.WAIT_GET_BOUNDARY,
+                    indexer.buildEndpoints().thenCompose(range ->
+                            recordStore.getPrimaryKeyBoundaries(range.getLow(), range.getHigh()).asList()));
+
+            logger.info("The boundary primary keys are " + boundaryPrimaryKeys);
+
+            commit(context);
+        }
+
+        int boundaryPrimaryKeysSize = boundaryPrimaryKeys.size();
+        assertTrue(boundaryPrimaryKeysSize > 2,
+                "the test is meaningless if the records are not across boundaries");
+        assertThat( boundaryPrimaryKeys.get(0), greaterThanOrEqualTo(Tuple.from(-25L * 39)));
+        assertThat( boundaryPrimaryKeys.get(boundaryPrimaryKeysSize - 1), lessThanOrEqualTo(Tuple.from(24L * 39)));
+        assertEquals(boundaryPrimaryKeys.stream().sorted().distinct().collect(Collectors.toList()), boundaryPrimaryKeys,
+                "the list should be sorted without duplication.");
+        for (Tuple boundaryPrimaryKey : boundaryPrimaryKeys) {
+            assertEquals(1, boundaryPrimaryKey.size(), "primary keys should be a single value");
+        }
+
+
+        // Test splitIndexBuildRange.
+        assertEquals(1, indexer.splitIndexBuildRange(Integer.MAX_VALUE, Integer.MAX_VALUE).size(),
+                "the range is not split when it cannot be split to at least minSplit ranges");
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize, indexer);
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize + 1, indexer); // Exactly one range for each split
+        checkSplitIndexBuildRange(boundaryPrimaryKeysSize / 2, boundaryPrimaryKeysSize + 2, indexer);
+
+        indexer.close();
+    }
+
+    private void checkSplitIndexBuildRange(int minSplit, int maxSplit, OnlineIndexer indexer) throws Exception {
+        List<Pair<Tuple, Tuple>> splitRanges = indexer.splitIndexBuildRange(minSplit, maxSplit);
+
+        assertThat(splitRanges.size(), greaterThanOrEqualTo(minSplit));
+        assertThat(splitRanges.size(), lessThanOrEqualTo(maxSplit));
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, TEST_SPLIT_HOOK);
+
+            // Make sure each enpoint is a valid primary key.
+            splitRanges.forEach(range -> {
+                recordStore.loadRecord(range.getLeft());
+                recordStore.loadRecord(range.getRight());
```

Comment:
`loadRecord` returns `null` if the record is absent, so I don't think this is testing anything. Maybe `assertTrue(recordStore.recordExists(foo))`?